src/Liquidator.sol

```diff
@@ -367,88 +372,76 @@ contract Liquidator is Owned, ILiquidator {
+        // Check if the account is being auctioned.
         if (!auctionInformation_.inAuction) revert Liquidator_NotForSale();
 
-        _knockDown(account, auctionInformation_);
+        bool success = _endAuction(account, auctionInformation_);
+        if (!success) revert Liquidator_EndAuctionFailed();
     }
 
     /**
-     * @notice Ends an auction when an Account has remaining debt and is healthy.
-     * @param account The account to end the liquidation for.
-     * @param auctionInformation_ The struct containing all the info of that specific auction.
+     * @notice Ends an auction and settles the liquidation.
+     * @param account The contract address of the account in liquidation.
+     * @param auctionInformation_ The struct containing all the auction information.
+     * @dev There are four different conditions on which an auction can be successfully ended.
+     * This function will check three of the four conditions (the fourth is already checked in the bid-function):
+     *  1) The Account is back in a healthy state (collateral value is equal or bigger than the used margin).
+     *  2) There are no remaining assets in the Account left to sell.
+     *  3) The Auction did not finish within the cutoff-period.
+     *  4) All open debt was repaid (not checked within this function).
+     * @dev If the third condition is met, an emergency process is triggered.
+     * The auction will be stopped and the remaining assets of the Account will be transferred to the Liquidator owner.
+     * The tranches of the liquidity pool will pay for the bad debt.
+     * The protocol will sell/auction the assets manually to recover the debt.
+     * The protocol will later "donate" these proceeds back to the
+     * impacted Tranches, this last step is not enforced by the smart contracts.
+     * While this process is not fully trustless, it is the only way to solve an extreme unhappy flow,
+     * where an auction did not end within cutoffTime (due to market or technical reasons).
      */
-    function _knockDown(address account, AuctionInformation storage auctionInformation_) internal {
-        // Set the inAuction flag to false.
-        auctionInformation[account].inAuction = false;
-
-        (bool success,,) = IAccount(account).isAccountHealthy(0, 0);
-        if (!success) revert Liquidator_AccountNotHealthy();
+    function _endAuction(address account, AuctionInformation storage auctionInformation_)
+        internal
+        returns (bool success)
+    {
+        // Stop the auction, this will prevent any possible reentrance attacks.
+        auctionInformation_.inAuction = false;
 
+        // Cache variables.
         uint256 startDebt = auctionInformation_.startDebt;
+        address creditor = auctionInformation_.creditor;
+
+        uint256 collateralValue = IAccount(account).getCollateralValue();
+        uint256 usedMargin = IAccount(account).getUsedMargin();
```

Comment:
can be merged in accounts?

Reply:
Yes, can be done, would add it to https://arcadiafinance.atlassian.net/browse/AF-376